frontend/src/components/pages/CreateReview/CreateReview.tsx

```diff
@@ -163,17 +163,34 @@ const CreateReview = ({ id }: CreateReviewProps): React.ReactElement => {
     if (review !== "" || reviewerByline !== "" || books.length !== 0) {
       // publish review
       if (authenticatedUser?.id) {
-        reviewAPIClient
-          .publishReview({
-            body: review,
-            byline: reviewerByline,
-            featured: featured === "1",
-            createdBy: parseInt(authenticatedUser?.id, 10),
-            publishedAt: new Date().getTime(),
-            books: mapBookToBookRequest(books),
-            tags: [],
-          })
-          .then((response) => {
+        const book = {
+          body: review,
+          byline: reviewerByline,
+          featured: featured === "1",
+          createdBy: parseInt(authenticatedUser?.id, 10),
+          publishedAt: new Date().getTime(),
+          books: mapBookToBookRequest(books),
+          tags: [],
+        };
+        if (id) {
+          reviewAPIClient
```

Comment:
(minor) It might be cleaner to avoid duplicating this code, ie do something like:
```js
const f = id ? reviewAPIClient.publishEditedReview : reviewAPIClient.publishCreatedReview
try {
  await f(book)
} catch(err) {...}

Reply:
both functions take different parameters though

Reply:
You could maybe put the code in the `.then` in a helper function maybe? Like `publishEditedReview(..).then(handlePublishResponse`. Up to you, this is a pretty minor nitpick

Reply:
I'm not sure I understand @susheelk